testutil/containschecker.go

```diff
@@ -179,39 +179,56 @@ func (c *deepUnsortedMatchChecker) Check(params []interface{}, _ []string) (bool
 		return false, fmt.Sprintf("containers are of different types: %s != %s", container1.Kind(), container2.Kind())
 	}
 
-	if container1.Kind() != reflect.Map && container1.Kind() != reflect.Slice && container1.Kind() != reflect.Array {
-		return false, fmt.Sprintf("'%s' is not a supported type: must be slice, array, map or nil", container1.Kind().String())
+	switch container1.Kind() {
+	case reflect.Array, reflect.Slice:
+		return deepSequenceMatch(container1, container2)
+	case reflect.Map:
+		return deepMapMatch(container1, container2)
+	default:
+		return false, fmt.Sprintf("'%s' is not a supported type: must be slice, array or map", container1.Kind().String())
 	}
+}
 
-	if container1.Type().Comparable() && params[0] == params[1] {
-		return true, ""
+func deepMapMatch(container1, container2 reflect.Value) (bool, string) {
+	if valid, output := validateContainerTypesAndLengths(container1, container2); !valid {
+		return false, output
 	}
 
-	if container1.Len() != container2.Len() {
-		return false, fmt.Sprintf("containers have different lengths: %d != %d", container1.Len(), container2.Len())
+	switch container1.Type().Elem().Kind() {
+	case reflect.Slice, reflect.Array, reflect.Map:
+	// only run the unsorted match if the map values are containers
+	default:
+		if !reflect.DeepEqual(container1.Interface(), container2.Interface()) {
+			return false, "maps don't match"
+		}
 	}
 
-	switch container1.Kind() {
-	case reflect.Array, reflect.Slice:
-		return deepSequenceMatch(container1, container2)
+	for _, key := range container1.MapKeys() {
+		el1 := container1.MapIndex(key)
+		el2 := container2.MapIndex(key)
 
-	case reflect.Map:
-		map1 := container1.Interface()
-		map2 := container2.Interface()
-		if !reflect.DeepEqual(map1, map2) {
-			return false, "maps don't match"
+		absent := el2 == reflect.Value{}
+		if absent {
+			return false, fmt.Sprintf("key %q from one map is absent from the other map", key)
 		}
 
-		return true, ""
-
-	default:
-		return false, fmt.Sprintf("%T is not a supported container. Must be a slice, an array or a map", container1)
+		switch el1.Kind() {
+		case reflect.Array, reflect.Slice:
+			return deepSequenceMatch(el1, el2)
+		case reflect.Map:
+			return deepMapMatch(el1, el2)
+		}
```

Comment:
This looks wrong: you are inside a loop, you should not return but rather do
```suggestion
			equal, msg = deepSequenceMatch(el1, el2)
		case reflect.Map:
			equal, msg = deepMapMatch(el1, el2)
		}
		if !equal {
			return equal, msg
		}
	}
```

Reply:
You're right, my mistake. Fixed it now, thanks!